cypress/integration/admin/pageBuilder/blocks/pageBlocks.cy.ts

```diff
@@ -0,0 +1,57 @@
+import { nanoid } from "nanoid";
+import { customAlphabet } from "nanoid";
+
+context("Blocks Page", () => {
+    let tokenStorage;
+    const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz");
+    const blockCategoryName = nanoid(10); // Generate a random 10-character lowercase string
+    const slugName = nanoid(10); // Generate another random 10-character lowercase string
+
+    beforeEach(() => cy.login());
+    beforeEach(() => cy.pbDeleteBlocks());
+    beforeEach(() => cy.pbAllDeleteBlockCategories());
+
+    it("Should be able to create a block category and a block and then edit, duplicate and delete it", () => {
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.findByTestId("pb-blocks-list-new-block-btn").click();
+        cy.findByTestId("pb-blocks-list-new-block-category-btn").click();
+
+        cy.findByRole("textbox", { name: "Name" }).type(blockCategoryName);
+        cy.findByRole("textbox", { name: "Slug" }).type(slugName);
+        cy.findByRole("textbox", { name: "Description" }).type(blockCategoryName);
+        cy.findByTestId("pb-block-categories-form-save-block-category-btn").click();
+
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.findByTestId("pb-blocks-list-new-block-btn").click();
+        cy.findByText(slugName).click();
+        cy.findByTestId("pb-blocks-editor-save-changes-btn").click();
+        cy.findByTestId("pb-blocks-list-new-block-btn").should("exist");
+
+        cy.visit("/page-builder/block-categories");
+        cy.findByPlaceholderText("Search block categories").should("exist");
+        cy.contains(blockCategoryName).click();
+        cy.findByRole("textbox", { name: "Description" }).type(blockCategoryName + "1");
+        cy.findByTestId("pb-block-categories-form-save-block-category-btn").click();
+        cy.findByPlaceholderText("Search block categories", { timeout: 10000 }).should("exist");
+
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.contains(blockCategoryName).click();
+        cy.findByTestId("pb-blocks-list-block-duplicate-btn").eq(0).click();
```

Comment:
When duplicating the block, let's see if the block can be seen in the list of blocks too.

Reply:
This one is still pending. Let's see if the block can be found in the list of blocks.